controller-server/src/main/java/com/yahoo/vespa/hosted/controller/tenant/Tenant.java

```diff
@@ -0,0 +1,79 @@
+// Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.hosted.controller.tenant;
+
+
+import com.yahoo.config.provision.TenantName;
+import com.yahoo.vespa.athenz.api.AthenzDomain;
+
+import java.util.Objects;
+
+/**
+ * A tenant in hosted Vespa.
+ *
+ * @author mpolden
+ */
+public abstract class Tenant {
```

Comment:
I don't like the `.type()`, `.isAthenz()`, `.isUser()`, `in(AthenzDomain)` and `isUser(String)` members. The type is already modelled through inheritance, so I don't see any strong arguments for having an explicit `Type` enum. I prefer having these methods replaced with use of `instanceof` + casting.

Reply:
Agreed, I'll replace it with explicit casts.

Reply:
PTAL